x/concentrated-liquidity/swapstrategy/zero_for_one.go

```diff
@@ -77,6 +81,21 @@ func (s zeroForOneStrategy) ComputeSwapWithinBucketOutGivenIn(sqrtPriceCurrent,
 		sqrtPriceNext = math.GetNextSqrtPriceFromAmount0InRoundingUp(sqrtPriceCurrent, liquidityBigDec, amountZeroInRemainingLessSpreadReward)
 	}
 
+	// The code snippet below deals with an edge case stemming from rounding.
+	// If the computed sqrtPriceNext is 1 BigDec ULP away from the sqrtPriceTarget AND
+	// amount zero in needed to reach target is 1 unit greater than the amount zero remaining after spread rewards,
+	// we deem this as a rounding error and make sqrtPriceNext equal to sqrtPriceTarget
+	// while consuming all amount in actually remaining.
+	//
+	// Without this change, we reach an infinite loop swap condition where we try to swap until the target but fail to consume
+	// any amount in to advance by only 1 ULP.
+	//
+	// Changing the rounding behavior causes issues in other parts of the system where we end up overconsuming the final amount in.
+	if sqrtPriceNext.Sub(sqrtPriceTarget).Equal(smallestBigDec) && amountZeroIn.Sub(amountZeroInRemainingLessSpreadReward).Equal(oneBigDec) {
+		sqrtPriceNext = sqrtPriceTarget
+		amountZeroIn = amountZeroInRemainingLessSpreadReward
+	}
+
```

Comment:
Is a similar situation possible in the InGivenOut direction?

Reply:
In given out did not experience this in the fuzzer.

Since rounding functions and logic are different between the two, a similar situation might be impossible for in given out.

Since this is more of a UX issue (some swaps failing due to infinite loops) and not a safety issue while being undetected by fuzzer, I don't think it is worth a deep investigation